### DISCUSSION TITLE
Using system trust CA store?

### DISCUSSION THREAD

**Discussion by TianyiChen** [cesanta/mongoose#2883] [Ask questions, get answers] [answered]
According to my understanding, to establish a TLS connection as client, a CA certificate needs to be explicitly selected. (or no verification at all). There doesn't seem to be an option to use system's trust CA store like calling [SSL_CTX_load_verify_locations](https://docs.openssl.org/master/man3/SSL_CTX_load_verify_locations/#description) with openssl. I think that would be handy. If this is deliberately not supported, is it due to performance/compatibility concerns?

**Comment by scaprile** [comment, marked as answer]
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).

Mongoose is embedded-oriented, in most usage scenarios there is no notion of "system".
https://mongoose.ws/documentation/tutorials/tls/
